kernel/src/main/java/org/kframework/kore/convertors/KILtoKORE.java

```diff
@@ -316,10 +316,36 @@ public Value applyAssoc(String assocOrig) {
                         } else if (it instanceof UserList) {
                             throw new AssertionError("Lists should have applied before.");
                         } else if (it instanceof Lexical) {
+                            String regex;
                             if (p.containsAttribute("regex"))
-                                items.add(RegexTerminal(p.getAttribute("regex")));
+                                regex = p.getAttribute("regex");
                             else
-                                items.add(RegexTerminal(((Lexical) it).getLexicalRule()));
+                                regex = ((Lexical) it).getLexicalRule();
+                            String precede = "#";
+                            if (regex.startsWith("(?<!")) { // find the precede pattern in the beginning: (?<!X)
+                                int depth = 1;
+                                for (int i = 1; i < regex.length(); i++) {
+                                    if (regex.charAt(i) == '\\') {
+                                        i++;
+                                        continue;
+                                    }
+                                    if (regex.charAt(i) == '(') depth++;
+                                    if (regex.charAt(i) == ')') depth--;
+                                    if (depth == 0) {
+                                        precede = regex.substring("(?<!".length(), i);
+                                        regex = regex.substring(i + 1);
+                                        break;
+                                    }
+                                }
+                            }
+                            String follow = "#";
+                            int followIndex = regex.indexOf("(?<!");
```

Comment:
since this is supposed to be negative lookahead, I believe you are supposed to drop the "<"